### PR TITLE
feat: add upgrade handler `v2.2.0`

### PR DESCRIPTION
**Pull request by atheeshp**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/claim/keeper/claim.go [hunk 1]
```diff
@@ -73,6 +74,28 @@ func (k Keeper) SetClaimRecord(ctx sdk.Context, claimRecord types.ClaimRecord) e
...
+	}
+
+	// if the record found in the state we no need to check the entries present in the claimRecord
+	existingClaimRecord, err := k.GetClaimRecord(ctx, addr)
+	if err != nil {
+		return err
+	}
+
+	if len(existingClaimRecord.ClaimableAmount) != 0 {
+		existingClaimRecord.ClaimableAmount[0].Amount = existingClaimRecord.ClaimableAmount[0].Amount.QuoRaw(5700).MulRaw(9322)
```
Comment (anilCSE) [inline x/claim/keeper/claim.go:91]:
```suggestion
		existingClaimRecord.ClaimableAmount[0].Amount = existingClaimRecord.ClaimableAmount[0].Amount.QuoRaw(3800).MulRaw(9322)
```

**Review by anilCSE — APPROVED** [review]
lgtm

File: app/upgrades/v2.2.0/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	pva := vestingtypes.NewPeriodicVestingAccount(authtypes.NewBaseAccount(vestingAcc, nil, ak.GetNextAccountNumber(ctx), 0),
+		amount,
+		1784905200,
+		genVestingPeriods(),
+	)
+	ak.SetAccount(ctx, pva)
+
+	if err := dk.DistributeFromFeePool(ctx, amount, vestingAcc); err != nil {
+		return err
+	}
```
Comment (aleem1314) [inline app/upgrades/v2.2.0/upgrade.go:69]: We have same logic inside v1 upgrade handler. Lets remove `v1` upgrade handler.

**Review by aleem1314 — APPROVED** [review]
Pre approving

File: app/upgrades/v2.2.0/upgrade.go [hunk 3]
```diff
@@ -56,7 +64,51 @@ func ExecuteProposal(ctx sdk.Context, dk distribution.Keeper, bk bank.Keeper, ak
...
+
+	// send the added balances from airdrop account to claim module account
+	if err := bk.SendCoinsFromAccountToModule(ctx, airdropAccAddr, claimtypes.ModuleName, amount); err != nil {
+		return err
+	}
+	ctx.Logger().Info(fmt.Sprintf("sent coins: %s from airdrop account to claim module account", amount.String()))
+
+	params := ck.GetParams(ctx)
+	params.AirdropEnabled = true
+	params.AirdropStartTime = time.Date(2023, 10, 12, 15, 0, 0, 0, time.UTC) // (dd/mm/yyyy: 16/10/2023, 15:00)
```
Comment (aleem1314) [inline app/upgrades/v2.2.0/upgrade.go:106]: which one is correct(comment  or airdrop-start-time)?

Comment (atheeshp) [reply app/upgrades/v2.2.0/upgrade.go:106]: @anilCSE mentioned in my old PR the actual date should be from 23/10/2023.

Comment (atheeshp) [inline app/upgrades/v2.2.0/upgrade.go:106]:
```suggestion
	params.AirdropStartTime = time.Date(2023, 10, 23, 15, 0, 0, 0, time.UTC) // (dd/mm/yyyy: 23/10/2023, 15:00UTC)
```

**Review by anilCSE — APPROVED** [review]

**Pull request merged**